pandas/core/accessor.py

```diff
@@ -107,7 +109,9 @@ def f(self, *args, **kwargs):
                 setattr(cls, name, f)
 
 
-def delegate_names(delegate, accessors, typ, overwrite=False):
+def delegate_names(
+    delegate, accessors: Union[List[str], Set[str]], typ: str, overwrite: bool = False
```

Comment:
@simonjayhawkins is there some container-ish thing we would use for `Union[List[foo], Set[foo]]`?  Also do we have a convention about annotating return from `__init__`?